scripts/watch_helpers/__init__.py

```diff
@@ -60,6 +62,23 @@ def get_client(dry_run: bool = False):
     return create_watch_client()
 
 
+def handle_posts(iter_func, collection, dry_run, *args, **kwargs):
+    client = get_client(dry_run)
+    with concurrent.futures.ThreadPoolExecutor(max_workers=32) as executor:
+        future_to_url = {
+            executor.submit(client.watch.post_stac_file, url, collection): url
+            for url in iter_func(*args, **kwargs)
+        }
+        for future in concurrent.futures.as_completed(future_to_url):
+            url = future_to_url[future]
+            try:
+                _ = future.result()
+            except Exception as exc:
+                print('%r generated an exception: %s' % (url, exc))
+            else:
+                print(f'Succeeded: {url}')
```

Comment:
@mcovalt, would you please take a look at this... I'm getting request timeouts with this. I suspect n_workers=32 isn't the right way to configure this?

Reply:
Request timeouts make me think the issue is now on the backend. We should be able to handle that many requests. We can add a PostgreSQL connection pooler and look into gevent.